fmt.sh

```diff
@@ -105,18 +113,20 @@ echo "\
 # Check for correct copyrights
 run dade-copyright-headers "$dade_copyright_arg" .
 
-# We have a Bazel test that is meant to run HLint, but we're a little sceptical of it
-# If we get this far, but hlint fails, that's a problem we should fix
-function bad_hlint() {
-  echo "UNEXPECTED HLINT FAILURE: The Bazel rules should have spotted this, please raise a GitHub issue"
-}
-trap bad_hlint EXIT
-for dir in daml-assistant libs-haskell compiler release language-support; do
-  run pushd "$dir"
-  run hlint --git -j4
-  run popd
-done
-trap - EXIT
+if [ "$hlint_diff" = "true" ]; then
+    changed_haskell_files="$(git diff --name-only origin/master | grep '.hs$')"
+    if [ "" != "$changed_haskell_files" ]; then
+        hlint -j4 $changed_haskell_files
+    fi
+else
+    # We have a Bazel test that is meant to run HLint, but we're a little
+    # sceptical of it If we get this far, but hlint fails, that's a problem we
+    # should fix
+    if ! hlint --git -j4; then
```

Comment:
Is the `cd` for the different directories no longer necessary?

Reply:
I don't know why it was there; as far as I can tell it was never necessary, hlint tests all the `.hs` files it can find.

Reply:
alright then let’s kill it.